scalameta/trees/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala

```diff
@@ -414,7 +414,12 @@ object TreeSyntax {
         case t: Defn.Var       => s(w(t.mods, " "), kw("var"), " ", r(t.pats, ", "), t.decltpe, " ", kw("="), " ", t.rhs.map(s(_)).getOrElse(s(kw("_"))))
         case t: Defn.Type      => s(w(t.mods, " "), kw("type"), " ", t.name, t.tparams, " ", kw("="), " ", t.body)
         case t: Defn.Class     => s(w(t.mods, " "), kw("class"), " ", t.name, t.tparams, w(" ", t.ctor, t.ctor.mods.nonEmpty), templ(t.templ))
-        case t: Defn.Trait     => s(w(t.mods, " "), kw("trait"), " ", t.name, t.tparams, templ(t.templ))
+        case t: Defn.Trait     =>
+          if (!dialect.allowTraitParameters && t.ctor.mods.nonEmpty) {
```

Comment:
Just minor nitpick, but I try to keep the happy path first

```diff
-          if (!dialect.allowTraitParameters && t.ctor.mods.nonEmpty) { 
+          if (dialect.allowTraitParameters || t.ctor.mods.isEmpty) { 
```

It flows better in my head for some reason, but maybe it's just me :/